department-of-veterans-affairs/content-build
# Vamc 15446 spike vamc police data page

## Description

relates to https://github.com/department-of-veterans-affairs/va.gov-cms/issues/15446

Creates a new process for queryType: "curl" in static-data-files

This allows an arbitrarily large number of files to be downloaded via curly/node-libcurl GET and a postProcess step that receives the data in an array and can process the resulting data in any manner needed. 

It uses libcurl because node-fetch has problems using file:// URLs which we need for the police instance. 

The police postProcess step writes the files to temporary files and reads them in with csvtojson to later process (this ticket only starts the structure, since it is unknown what the format of the CSVs will be at the moment).

Changes were made to the static file creation, in that if any of the files are not present, all are generated. Previously only a test was done to determine if the directory exists, but now that there are multiple processes generating files, it seemed appropriate to convert to a per-file test. 

## Testing done & Screenshots

Added cypress testing for the existence of all JSON static data files.


## QA steps


What needs to be checked to prove it didn't break any related things?  
What variations of circumstances (users, actions, values) need to be checked?  

1. Check if https://web-w1a5evfuonymjntciz3pduj1ofhepwky.demo.cms.va.gov/data/cms/vamc-facility-supplemental-status.json exists
2. Check if https://web-w1a5evfuonymjntciz3pduj1ofhepwky.demo.cms.va.gov/data/cms/vamc-ehr.json exists
3. Check if https://web-w1a5evfuonymjntciz3pduj1ofhepwky.demo.cms.va.gov/data/cms/vamc-police.json exists
    1. it is an empty object because I didn't want to put fake data into production. 

## Acceptance criteria

- [x] A skeleton is started for the general file structure pending any missing information. (e.g., if the output has to be JSON, the Ruby requirements are documented)

## Definition of done
- [x] Documentation has been updated, if applicable (added testing with comments, looking for other locations for other notes about static data files)
